Watchman.Discord/Areas/Responses/Controllers/ResponsesController.cs

```diff
@@ -111,27 +111,27 @@ public async Task RemoveResponse(DiscordRequest request, Contexts contexts)
 
         [AdminCommand]
         [DiscordCommand("responses")]
-        public Task Responses(DiscordRequest request, Contexts contexts)
+        public async Task Responses(DiscordRequest request, Contexts contexts)
         {
-            var args = request.Arguments?.ToArray();
+            var arg = request.Arguments?.FirstOrDefault();
             string commandArgument = null;
 
-            if (args?.Length == 0)
+            if (arg == null)
             {
                 commandArgument = "all";
             }
-
-            else if (args[0].Name != "all" && args[0].Name != "default" && args[0].Name != "custom")
+            else if (arg.Name != "all" && arg.Name != "default" && arg.Name != "custom")
             {
-                throw new InvalidArgumentsException();
+                throw new InvalidArgumentsException("-all -default -custom");
             }
 
-            commandArgument = commandArgument ?? args[0].Name;
+            commandArgument = commandArgument ?? arg.Name;
             var messageService = _messagesServiceFactory.Create(contexts);
-            var allResponses = new ResponsesDatabase(_queryBus, contexts.Server.Id);
-            var responsesMessageService = new ResponsesMessageService(allResponses, messageService);
+            var embedMessageSplittingService = new EmbedMessageSplittingService(messageService);
+            var allResponses = new ResponsesGetterService(_queryBus);
+            var responsesMessageService = new ResponsesMessageService(allResponses, embedMessageSplittingService, contexts.Server.Id);
```

Comment:
Właśnie chodzi o to, żeby nie przekazywać serwisów do konstruktorów
Na tym polega IoC i autofac, żeby ręcznie nie tworzyć instancji
Po prostu przyjmuj te serwisy w konstruktorze i tyle - dokładnie to wystarczy sam ResponsesMessageService

Reply:
czyli co mam zrobić?

Reply:
Mam ten ResponsesMessageService przyjąć przez konstruktor klasy ResponsesController?

Reply:
tak i przypisać do prywatnego pola readonly tak jak inne serwisy w controllerach